account_invoice_actual_date/models/sale_order.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# Licence AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3.0).
+
+from odoo import fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _prepare_invoice(self):
+        invoice_vals = super()._prepare_invoice()
+        if self.picking_ids:
+            if self.picking_ids[0].actual_date:
+                invoice_vals["invoice_date"] = self.picking_ids[0].actual_date
+            elif self.picking_ids[0].date_done:
+                invoice_vals["invoice_date"] = fields.Datetime.context_timestamp(
+                    self.with_context(tz=self.env.company.partner_id.tz),
```

Comment:
Just take the company from the record being worked on.
```suggestion
                    self.with_context(tz=self.company_id.partner_id.tz),
```